app/backend/approaches/retrievethenread.py

```diff
@@ -114,18 +117,15 @@ async def run(
 
         # Process results
         sources_content = self.get_sources_content(results, use_semantic_captions, use_image_citation=False)
+        current_model = self.hf_model if self.hf_model else self.chatgpt_model
 
-        # Append user message
-        content = "\n".join(sources_content)
-        user_content = q + "\n" + f"Sources:\n {content}"
-
-        response_token_limit = 1024
-        updated_messages = build_messages(
-            model=self.chatgpt_model,
-            system_prompt=overrides.get("prompt_template", self.system_chat_template),
-            few_shots=[{"role": "user", "content": self.question}, {"role": "assistant", "content": self.answer}],
-            new_user_content=user_content,
-            max_tokens=self.chatgpt_token_limit - response_token_limit,
+        # Load the Prompty object
+        prompty_path = SUPPORTED_MODELS.get(current_model) / "ask.prompty"
+        prompty = Prompty.load(source=prompty_path)
```

Comment:
Do what you think, but I would just put it that way ^^
```suggestion
        prompty = Prompty.load(source=SUPPORTED_MODELS.get(current_model) / "ask.prompty")
```